hexa/connections/tests/test_dhis2connection.py

```diff
@@ -0,0 +1,291 @@
+            json={"organisationUnits": org_units},
+            status=200,
+        )
+        dhis2 = get_client_by_slug("dhis2-connection-1", self.USER_JIM)
+        metadata = get_dhis2_metadata(dhis2, "organisation_units", fields="id,name")
+        self.assertIsNotNone(metadata)
+
+    @responses.activate
+    def test_dhis2_org_units_by_groups(self):
+        responses._add_from_file(
+            Path("hexa", "connections", "tests", "fixtures", "dhis2_init.yaml")
+        )
+        responses.add(
+            responses.GET,
+            "http://127.0.0.1:8080/api/organisationUnits?fields=id%2Cname&filter=organisationUnitGroups.id%3Ain%3A%5BoDkJh5Ddh7d%5D&pageSize=1000",
+            json={"organisationUnits": org_units},
+            status=200,
+        )
+        dhis2 = get_client_by_slug("dhis2-connection-1", self.USER_JIM)
+        metadata = get_dhis2_metadata(
+            dhis2,
+            "organisation_units",
+            filter="organisationUnitGroups.id:in:[oDkJh5Ddh7d]",
+            fields="id,name",
+        )
+        self.assertIsNotNone(metadata)
+
+    @responses.activate
+    def test_dhis2_org_unit_groups(self):
+        responses._add_from_file(
+            Path("hexa", "connections", "tests", "fixtures", "dhis2_init.yaml")
+        )
+        responses.add(
+            responses.GET,
+            "http://127.0.0.1:8080/api/organisationUnitGroups?fields=id%2Cname&pageSize=1000",
+            json={"organisationUnitGroups": org_units_groups},
+            status=200,
+        )
+        dhis2 = get_client_by_slug("dhis2-connection-1", self.USER_JIM)
+        metadata = get_dhis2_metadata(
+            dhis2, type="organisation_unit_groups", fields="id,name"
+        )
+        self.assertIsNotNone(metadata)
+
+    @responses.activate
+    def test_dhis2_org_unit_levels(self):
+        responses._add_from_file(
+            Path("hexa", "connections", "tests", "fixtures", "dhis2_init.yaml")
+        )
+        responses.add(
+            responses.GET,
+            "http://127.0.0.1:8080/api/filledOrganisationUnitLevels",
+            json=org_units_levels,
+            status=200,
+        )
+        dhis2 = get_client_by_slug("dhis2-connection-1", self.USER_JIM)
+        metadata = get_dhis2_metadata(dhis2, "organisation_unit_levels")
+        self.assertIsNotNone(metadata)
+
+    @responses.activate
+    def test_dhis2_datasets(self):
+        responses._add_from_file(
+            Path("hexa", "connections", "tests", "fixtures", "dhis2_init.yaml")
+        )
+        responses.add(
+            responses.GET,
+            "http://127.0.0.1:8080/api/dataSets?fields=id%2Cname&pageSize=1000",
+            json={"dataSets": datasets},
+            status=200,
+        )
+        dhis2 = get_client_by_slug("dhis2-connection-1", self.USER_JIM)
+        metadata = get_dhis2_metadata(dhis2, "datasets", fields="id,name")
+        self.assertIsNotNone(metadata)
```

Comment:
Maybe we could get some value, at least for some tests, of asserting the state of metadata and not only checking that it's not `None` ?